backend/packages/Upgrade/src/api/services/ExperimentAssignmentService.ts

```diff
@@ -505,6 +521,93 @@ export class ExperimentAssignmentService {
     }
   }
 
+  private async experimentLevelExclusionInclusion(experiments: Experiment[], experimentUser: ExperimentUser): Promise<Experiment[]> {
+
+    let expLevelFilteredExperiments = [];
+
+    const explicitExperimentIndividualExclusionData = (await this.explicitExperimentIndividualExclusionRepository
+      .findAllUsers())
+      .filter(element => element.userId === experimentUser.id)
+      .map(element => ({ userId: element.userId,  experimentId: element.experiment.id }));
+
+    const explicitExperimentIndividualInclusionData = (await this.explicitExperimentIndividualInclusionRepository
+      .findAllUsers())
+      .filter(element => element.userId === experimentUser.id)
+      .map(element => ({ userId: element.userId,  experimentId: element.experiment.id }));
+
+    const explicitExperimentGroupExclusionData = (await this.explicitExperimentGroupExclusionRepository
+      .findAllGroups())
+      .map((element) => ({ groupId: element.groupId,  type: element.type, experimentId: element.experiment.id }));
+
+    const explicitExperimentGroupInclusionData = (await this.explicitExperimentGroupInclusionRepository
+      .findAllGroups())
+      .map((element) => ({ groupId: element.groupId,  type: element.type, experimentId: element.experiment.id }));
```

Comment:
Can you `Promise.all` instead of awaiting them individually

Reply:
I have updated the code. Thank you.